ironfish/src/network/peerNetwork.ts

```diff
@@ -589,14 +646,29 @@ export class PeerNetwork {
         responseMessage = new CannotSatisfyRequest(rpcId)
       }
 
-      // TODO(daniel) if this is a response with transactions mark transaction hashes as seen by peer
-      peer.send(responseMessage)
+      const sent = peer.send(responseMessage)
+      if (
+        sent &&
+        responseMessage instanceof PooledTransactionsResponse &&
+        peer.state.identity
+      ) {
+        for (const serializedTransaction of responseMessage.transactions) {
+          const hash = new Transaction(serializedTransaction).hash()
+          this.markKnowsTransaction(hash, peer.state.identity)
+        }
+      }
     } else {
       const request = this.requests.get(rpcId)
       if (request) {
         request.resolve({ peerIdentity, message: rpcMessage })
-      } else {
-        this.logger.debug(`Dropping response to unknown request ${rpcId}`)
```

Comment:
Maybe we could put this back and add it to an `else` after the `if (rpcMessage instanceof PooledTransactionsResponse) {` call? I don't feel too strongly about it though

Reply:
I feel like RPC handling needs to be revisited so I think it's uncertain if we'll want this. I tried adding it and it would need to be 
```
} else if (!request) {
   this.logger.debug(`Dropping response to unknown request ${rpcId}`)
}
```

and it just seems kinda confusing